### PR TITLE
fix(cart): support cart entry volume price

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
We've ignored the volume price on cart entries so far. With this fix we show the cart unit price next to the (strikethrough) original item price. The prices are now formatted using the `SitePriceComponent`.

closes: [HRZ-3182](https://spryker.atlassian.net/browse/HRZ-3182)

[HRZ-3182]: https://spryker.atlassian.net/browse/HRZ-3182?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/cart/entry/src/entry.model.ts [hunk 1]
```diff
@@ -21,6 +21,11 @@ export interface CartEntryAttributes {
    */
   price?: number;
 
+  /**
+   * The item sales price that override the item sales price.
```
Comment (tolerants) [inline libs/domain/cart/entry/src/entry.model.ts]: Can you double check this description?

File: libs/domain/cart/entry/src/entry.model.ts [hunk 2]
```diff
@@ -21,6 +21,11 @@ export interface CartEntryAttributes {
    */
   price?: number;
```
Comment (tolerants) [inline libs/domain/cart/entry/src/entry.model.ts:25]: It looks like we are not using it anymore.

File: libs/domain/cart/entry/src/entry.component.ts [hunk 3]
```diff
@@ -60,7 +61,8 @@ export class CartEntryComponent
 
   @signalProperty({ type: Number }) quantity?: number;
   @property() key?: string;
-  @property({ type: Number }) price?: number;
+  @property({ type: Number }) itemPrice?: number;
+  @property({ type: Number }) entryPrice?: number;
```
Comment (tolerants) [inline libs/domain/cart/entry/src/entry.component.ts]: Snouldn't be `entryPrice` a part of entry model?

File: libs/domain/product/price/src/price.component.ts [hunk 4]
```diff
@@ -35,10 +28,44 @@ export class ProductPriceComponent extends ProductMixin(
...
 
-  protected $prices = computed(() => {
-    return this.formatPrices(this.$product()?.price);
+  protected $isDiscounted = computed(() => {
+    return (
+      (this.sales &&
+        typeof this.$salesPrice() === 'number' &&
+        this.sales < Number(this.$salesPrice())) ||
+      (typeof this.$salesPrice() === 'number' &&
+        typeof this.$salesPrice() === 'number' &&
```
Comment (tolerants) [inline libs/domain/product/price/src/price.component.ts]: Why we need to identical conditions?

**Review by tolerants — CHANGES_REQUESTED** [review]

File: libs/domain/product/price/src/price.component.ts (hunk 4 above)
Comment (tobi-or-not-tobi) [reply libs/domain/product/price/src/price.component.ts]: Thanks, that was wrong. I've actually updated the logic and aligned e2e.

Comment (tobi-or-not-tobi) [reply libs/domain/product/price/src/price.component.ts]: Also renamed the `SitePriceComponent`, as it had a flaw.

**Review by tolerants — DISMISSED** [review]

**Review by tolerants — DISMISSED** [review]

**Review by tolerants — DISMISSED** [review]

**Pull request merged**